### PR TITLE
Tesla tweak update to previous update and re-haul[buff]

### PR DESCRIPTION
**Pull request by Sindorman**
This Pr tweaks previous update

- Grounding rods now are vital for how many miniballs of Tesla can be contained without it going nuts and melt stuff. 1 rod = 4 miniballs. So with standard setup 4 rods = 16 miniballs max and that is around 4 million W energy.

- Each time Tesla melts coil/rod/emitter it will loose a single miniball and drop in energy.

- Tesla miniballs no longer shoot beams, instead they shoot main ball which shoot coils, accounting for energy produced by all miniballs.

- Tesla miniballs no longer zap. Instead they shoot main Tesla ball which shoots one single beamn, which accounts how much power each miniball produces.

- Tesla now looses more and faster energy with number of miniballs it has. Energy loss changes after 12 balls and after 16 miniballs.

- Each miniball produces 156250W

- Tesla Engine room is reduced in size, field is now 5x5.

- Tesla now moves smoothly tile-by-tile. No more *jumps*

```
All 17 balls energize for a second, sending their energy to the main ball, which redirects it at the nearest object!
The tesla coil melts down until ashes are left!

... 8 more melted rods later...
All 17 balls energize for a second, sending their energy to the main ball, which redirects it at the nearest object!
Grounding Rod melts down until ashes are left!

... 3 grounding rods later ...
All 15 balls energize for a second, sending their energy to the main ball, which redirects it at the nearest object!
The emitter melts down until ashes are left!
```

### CODE REVIEW

**Comment by Sindorman** [comment]
Feedback thread if someone wants https://forums.aurorastation.org/topic/11378-recent-tesla-nerfs-1-dismissal/

**Comment by Nickvr628** [comment]
If the goal is to nerf it, the station still does not draw anywhere near that amount of power, but this is a step in the right direction.

**Comment by Sindorman** [comment]
This is a buff, not a nerf. That is true, station draws 2 million W max, but this is an experimental engine, so it is normal for it to produce more power with right configuration.

File: code/modules/power/tesla/energy_ball.dm [hunk 1]
```diff
@@ -81,7 +84,7 @@
 
 	var/move_dir = 0
 	if(target && prob(75))
-		move_dir = get_dir(src,target)
+		move_dir = get_dir(src, target)
```
Comment (Karolis2011) [inline code/modules/power/tesla/energy_ball.dm]: How does this compile?

Comment (Sindorman) [reply code/modules/power/tesla/energy_ball.dm]: Why wouldn't it?

Comment (Sindorman) [reply code/modules/power/tesla/energy_ball.dm]: It doesn't care if you pass in loc or object.

File: code/modules/power/tesla/energy_ball.dm [hunk 2]
```diff
@@ -109,13 +112,13 @@
 		switch(z_move)
 			if(1)
 				visible_message(span("danger","\The [src] gravitates upwards!"))
-				forceMove(T)
+				zMove(UP)
 				visible_message(span("danger","\The [src] gravitates from below!"))
 			if(0)
-				forceMove(T)
+				walk_to(src, T, 0, speed)
```
Comment (skull132) [inline code/modules/power/tesla/energy_ball.dm]: psure there's an issue with `walk_to` that stops clean destroy. Do you have to use it? If you do, then make sure to cancel the `walk_to` in `energy_ball/Destroy()`. Unless it's already done at the atom level. 🤔

Comment (Sindorman) [reply code/modules/power/tesla/energy_ball.dm]: I do want to use it. I will add cancellation to `Destroy()`

**Pull request merged**